lib/Compiler.js

```diff
@@ -79,22 +77,29 @@ Watching.prototype._go = function() {
 	});
 };
 
+Watching.prototype._getStats = function(compilation) {
+	var stats = new Stats(compilation);
+	stats.startTime = this.startTime;
+	stats.endTime = new Date().getTime();
+	return stats;
+};
+
 Watching.prototype._done = function(err, compilation) {
 	this.running = false;
 	if(this.invalid) return this._go();
-	this.error = err || null;
-	this.stats = compilation ? compilation.getStats() : null;
-	if(this.stats) {
-		this.stats.startTime = this.startTime;
-		this.stats.endTime = new Date().getTime();
+
+	var stats = this._getStats(compilation);
```

Comment:
This line throws an error if there's no compilation.. e.g.: if `watch-run` plugin handlers call callback with an error, compilation will be null.

@sokra @timse